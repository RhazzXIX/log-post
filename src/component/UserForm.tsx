export default function UserForm({ version }: { version: string }) {
  return (
    <form action="" method="POST">
      <label htmlFor="username">
        Username:
        <input type="text" id="username" name="username" />
      </label>
      <label htmlFor="password">
        Password:
        <input type="password" name="password" id="password" />
      </label>
      {/* Logic to show password confirmation for version sign-up */}
      {version === "sign-up" && (
        <label htmlFor="passwordConfirmation">
          Confirm password:
          <input
            type="password"
            name="passwordConfirmation"
            id="passwordConfirmation"
          />
        </label>
      )}
      <button type="submit">Submit</button>
    </form>
  );
}
