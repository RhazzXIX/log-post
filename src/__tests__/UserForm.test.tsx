import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import UserForm from "@/component/UserForm";

describe("UserForm component", () => {
  it("Renders on screen", () => {
    const { container } = render(<UserForm version={"log-in"} />);
    expect(container).toMatchSnapshot();
  });
  it("Shows the confirm password input if the versions is sign-up", () => {
    const { rerender } = render(<UserForm version="log-in" />);

    expect(screen.queryByLabelText("Confirm password")).not.toBeInTheDocument();

    rerender(<UserForm version="sign-up" />);

    expect(screen.queryByLabelText("Confirm password:")).toBeInTheDocument();
  });
});
