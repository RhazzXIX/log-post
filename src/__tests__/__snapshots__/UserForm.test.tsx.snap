// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm component Renders on screen 1`] = `
<div>
  <form
    action=""
    id="userForm"
    method="POST"
  >
    <label
      for="username"
    >
      Username:
      <input
        id="username"
        minlength="3"
        name="username"
        type="text"
        value=""
      />
    </label>
    <label
      for="password"
    >
      Password:
      <input
        id="password"
        minlength="5"
        name="password"
        type="password"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
